File: in_Java/0257 Binary Tree Paths.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<String> binaryTreePaths(TreeNode root) {
        List<String> res = new ArrayList<>();
        if(root == null) return res;
        Stack<TreeNode> stn = new Stack<>();
        Stack<String> sts = new Stack<>();
        stn.push(root);
        sts.push("");
        while(!stn.isEmpty()){
            TreeNode cur = stn.pop();
            String now = sts.pop();
            if(cur.left == null && cur.right == null) res.add(now + cur.val);
            if(cur.left != null){
                stn.push(cur.left);
                sts.push(now + cur.val + "->");
            }
            if(cur.right != null){
                stn.push(cur.right);
                sts.push(now + cur.val + "->");
            }
        }
        return res;
    }
}

File: in_Python/0257 Binary Tree Paths.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def binaryTreePaths(self, root):
        """
        :type root: TreeNode
        :rtype: List[str]
        """
        st = []
        res = []
        if root == None:
            return res
        st.append([root,""])
        while st:
            n, s = st.pop()
            if n.left == None and n.right == None:
                res.append(s + str(n.val))
            if n.right != None:
                st.append([n.right, s + str(n.val) + '->'])
            if n.left != None:
                st.append([n.left, s + str(n.val) + '->'])
        return res